src/main/java/com/mathworks/ci/MatlabBuilder.java

```diff
@@ -408,9 +412,7 @@ private synchronized int execMatlabCommand(Run<?, ?> build, FilePath workspace,
             TaskListener listener, boolean isLinuxLauncher)
             throws IOException, InterruptedException {
         setEnv(build.getEnvironment(listener));
-        //Copy MATLAB scratch file into the workspace
-        copyMatlabScratchFileInWorkspace(MATLAB_RUNNER_RESOURCE, MATLAB_RUNNER_TARGET_FILE,
```

Comment:
Ideally we could just find a way to conditionally do this copy based on the test run mode. This will avoid the need to inject the knowledge of scratch file down in low level functions who are trying to determine the command based on -r and -batch.